### PR TITLE
finished milestone-2

### PR DESCRIPTION
**Pull request by tafodinho**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,8 +1,24 @@
 {
-    "extends": "airbnb",
+    "env": {
+        "browser": true,
+        "es6": true
+    },
+    "extends": [
```
Comment (zenott) [inline .eslintrc.json]: Let's add `plugin:react/recommended` here to have set up react related rules. Or better yet `airbnb` which also includes these rules.

File: src/containers/BookForm.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class BookForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    };
+  }
+
+  render() {
+    const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+    const options = categories.map((category, key) => (
```
Comment (zenott) [inline src/containers/BookForm.js]: Using index as key for lists in react is generally a [bad practice](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). We can just use `category` as key here as it is a unique identifier.

File: src/containers/BookList.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      </table>
+    </div>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  books: state.books,
+});
+BookList.propTypes = {
+  books: PropTypes.instanceOf(Array).isRequired,
```
[the remark this answers is not in the document]
Comment (tafodinho) [reply src/containers/BookList.js]: the linter cries that the array proptype is forbidden

**Comment by tafodinho** [comment]
Hi @zenott adding a state to the BookForm component is rather a milestone 3 tast as it is not required in milestone 2.

**Comment by zenott** [comment]
> Hi @zenott adding a state to the BookForm component is rather a milestone 3 tast as it is not required in milestone 2.

Hi @tafodinho ,

You are right, it's in milestone 3 indeed. We have to fix that in the review guidelines. Thanks for pointing it out.

File: src/components/Book.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable react/jsx-filename-extension */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Book = (props) => {
+  const { id, title, category } = props.book;
```
Comment (SunnyAsar) [inline src/components/Book.js]: de structure like this when its multiple layers:
```
const { book: { id, title, category} } = props;
```

**Comment by tafodinho** [comment]
Hi @SunnyAsar am kind of having a conflict between my text editor linting and that of stickler. my text edittor expects me not to put curcly braces around single function parameters. where's stickler expects the opposite. How can i do to fix this

**Pull request merged**